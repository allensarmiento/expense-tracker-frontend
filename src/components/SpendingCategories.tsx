import styled from 'styled-components';
import BannerTitle from './BannerTitle';
import BannerSection from './BannerSection';

const Categories = styled.div`
  flex: 1;
  background-color: #fff;
  border-radius: .4rem;
  overflow-y: auto;

  & > *:nth-child(odd) {
    background: var(--table-odd-bg-color);
  }
`;

const Category = styled.div`
  display: grid;
  grid-template-columns: minmax(20rem, max-content) minmax(10rem, 1fr);
  padding: .5rem 1rem;
`;

const CategoryName = styled.span`
  text-overflow: ellipses;
  overflow: hidden;
`;

const CategoryAmount = styled.span`
  text-overflow: ellipses;
  overflow: hidden;
`;

type CategoryMap = {
  [key: string]: number;
};

type SpendingCategoriesProps = {
  categories: CategoryMap;
};

const SpendingCategories = ({ categories }: SpendingCategoriesProps) => {
  return (
    <BannerSection>
      <BannerTitle>Spending Categories</BannerTitle>
      <Categories>
        {Object.keys(categories).map((category, key) => (
          <Category key={key}>
            <CategoryName>{category}</CategoryName>
            <CategoryAmount>${categories[category]}</CategoryAmount>
          </Category>
        ))}
      </Categories>
    </BannerSection>
  );
};

export default SpendingCategories;
