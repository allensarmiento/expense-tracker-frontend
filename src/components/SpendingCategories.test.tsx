import { shallow } from 'enzyme';
import SpendingCategories from './SpendingCategories';

it('renders without crashing', () => {
  expect(shallow(
    <SpendingCategories categories={{
      food: 20.00,
      rent: 1000.00
    }} />
  )).toMatchSnapshot();
});
