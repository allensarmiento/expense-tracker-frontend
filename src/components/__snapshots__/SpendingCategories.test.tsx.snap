// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<BannerSection>
  <BannerTitle>
    Spending Categories
  </BannerTitle>
  <styled.div>
    <styled.div
      key="0"
    >
      <styled.span>
        food
      </styled.span>
      <styled.span>
        $
        20
      </styled.span>
    </styled.div>
    <styled.div
      key="1"
    >
      <styled.span>
        rent
      </styled.span>
      <styled.span>
        $
        1000
      </styled.span>
    </styled.div>
  </styled.div>
</BannerSection>
`;
